### PR TITLE
feat(button): add states style

### PR DESCRIPTION
**Pull request by rachelbt**
still missing design tokens of colors to match the design

### CODE REVIEW

File: libs/shared/src/lib/sass/mixins/_states.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      background-color: var(--vvd-color-neutral-70);
+    }
+
+    //ghost & outlined
+    &:not(.appearance-filled) {
+      &:hover {
+        background-color: var(--vvd-color-neutral-10);
+      }
+      &:active {
+        background-color: var(--vvd-color-neutral-20);
```
Comment (yinonov) [inline libs/shared/src/lib/sass/mixins/_states.scss:18]: all variables should refer to scss consts

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/shared/src/lib/sass/mixins/_states.scss [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    }
+  }
+  $connotation: (
```
Comment (yinonov) [inline libs/shared/src/lib/sass/mixins/_states.scss:22]: we wont always have connotation, so maybe just have a default style if no connotation

File: libs/shared/src/lib/sass/mixins/_states.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        background-color: var(--vvd-color-#{$key}-90);
+      }
+
+      &:not(.appearance-filled) {
+        &:hover {
+          background-color: var(--vvd-color-#{$key}-10);
+        }
+        &:active {
+          background-color: var(--vvd-color-#{$key}-20);
+        }
```
Comment (yinonov) [inline libs/shared/src/lib/sass/mixins/_states.scss:45]: we need to map this our in the mixin
like in
https://github.com/Vonage/vivid-3/blob/6e76ede9b7208b95eb20eb4b3d9677ba094cdc65/libs/shared/src/lib/sass/mixins/connotation/_variables.scss#L9-L46

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/shared/src/lib/sass/mixins/_states.scss (hunk 2 above)
Comment (rachelbt) [reply libs/shared/src/lib/sass/mixins/_states.scss]: I added a default state

File: libs/shared/src/lib/sass/mixins/_states.scss (hunk 3 above)
Comment (rachelbt) [reply libs/shared/src/lib/sass/mixins/_states.scss]: added colors to the map and used them in the state mixin

**Review by yinonov — CHANGES_REQUESTED** [review]
Should we disable indication on a disabled/idle control?

**Comment by rachelbt** [comment]
> 

Right!! I've restricted the style when not idle. I don't think we need to target when disabled (as attribute) because it added the class (*-idle) anyway, right?

File: libs/components/src/lib/button/button.scss [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 @use '../../../../shared/src/lib/sass/mixins/connotation/config' with (
 	$subset: primary cta success alert,
-	$shades: diverse,
+	$shades: diverse contrast soft softer,
```
Comment (yinonov) [inline libs/components/src/lib/button/button.scss:7]: Softer? Softest? Final.final.design?

Comment (rachelbt) [reply libs/components/src/lib/button/button.scss:7]: is _light_ better?

Comment (yinonov) [reply libs/components/src/lib/button/button.scss:7]: yeap

Comment (yinonov) [reply libs/components/src/lib/button/button.scss:7]: muted, pale, dull, dim etc'...

Comment (rachelbt) [reply libs/components/src/lib/button/button.scss:7]: ohhhh - pale I like!

**Comment by yinonov** [comment]
@rachelbt run `npx beachball change` to create the change file. it's a `none` type

**Review by yinonov — DISMISSED** [review]

**Comment by yinonov** [comment]
> decided to control the states via appearance and not as a separated mixin

do we still need `libs/shared/src/lib/sass/mixins/_states.scss`?

**Pull request closed without merging**